testing/tests/DevExpress.ui.widgets.editors/validator.tests.js

```diff
@@ -448,6 +450,32 @@ QUnit.module('Events', {
         assert.deepEqual(params.brokenRule, expectedFailedValidationRule, 'Failed rules were passed');
     });
 
+    QUnit.test('Validated event should fire correctly after option runtime change', function(assert) {
+        const value = '';
+        const name = 'Login';
+        const expectedFailedValidationRule = { index: 0, type: 'required', isValid: false, message: 'Login is required', validator: {}, value: value };
+        const handler = sinon.stub();
+        const handlerAfterChange = sinon.stub();
+
+        const validator = this.fixture.createValidator({
+            name: name,
+            onValidated: handler,
+            validationRules: [{ type: 'required' }]
+        });
+        expectedFailedValidationRule.validator = validator;
+        this.fixture.stubAdapter.getValue.returns(value);
+
+        validator.validate();
+
+        expectedFailedValidationRule.validator.option('onValidated', handlerAfterChange);
+        this.fixture.stubAdapter.getValue.returns(value);
+
+        validator.validate();
+
+        assert.ok(handler.calledOnce, 'Validated handler should be called');
```

Comment:
We already have such assert:
https://github.com/DevExpress/DevExtreme/pull/11369/files#diff-dc05fa7fdaba59cd05c488434aca063dR434